File: sem3/dsa/sorting/insertion_sort.py
def insertion_sort(array):
    N = len(array)
    for i in range(N):
        for j in range(i, 0, -1):
            if array[j] < array[j-1]:
                exchange(j, j-1, array)
    return array

def exchange(i, j, array):
    temp = array[i]
    array[i] = array[j]
    array[j] = temp

print(insertion_sort([6,3,2,0,10,4,2]))

File: sem3/dsa/sorting/insertion_sort.c
#include <stdio.h>

void exchange(int i, int j, int array[]);

void insertion_sort(int array[], int N){
  for (int i=0; i<N; i++)
    for(int j=i; j>0; j--)
      if (array[j] < array[j-1])
        exchange(j, j-1, array);
}

void exchange(int i, int j, int array[]){
  int temp = array[i];
  array[i] = array[j];
  array[j] = temp;
}

int main(){
  int array[] = {5,3,0,4,2,5};
  int N = sizeof(array)/sizeof(int);
  insertion_sort(array, N);
  for (int i=0; i<N; i++){
    printf("%d ", array[i]);
  }
}
